fhir-server/src/main/java/au/csiro/pathling/sql/dates/TemporalArithmeticFunction.java

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright © 2018-2022, Commonwealth Scientific and Industrial Research
+ * Organisation (CSIRO) ABN 41 687 119 230. Licensed under the CSIRO Open Source
+ * Software Licence Agreement.
+ */
+
+package au.csiro.pathling.sql.dates;
+
+import static au.csiro.pathling.utilities.Preconditions.checkUserInput;
+
+import au.csiro.pathling.fhirpath.encoding.QuantityEncoding;
+import au.csiro.pathling.fhirpath.literal.QuantityLiteralPath;
+import au.csiro.pathling.sql.udf.SqlFunction2;
+import com.google.common.collect.ImmutableMap;
+import java.math.RoundingMode;
+import java.util.Calendar;
+import java.util.Map;
+import java.util.function.BiFunction;
+import java.util.function.Function;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import org.apache.spark.sql.Row;
+import org.apache.spark.sql.types.DataType;
+import org.apache.spark.sql.types.DataTypes;
+import org.hl7.fhir.r4.model.BaseDateTimeType;
+import org.hl7.fhir.r4.model.Quantity;
+
+public abstract class TemporalArithmeticFunction<IntermediateType extends BaseDateTimeType> implements
+    SqlFunction2<String, Row, String> {
+
```

Comment:
I think that the oracle convention for the names of type parameters of generic types is a single upper case letter.
This is probably a matter of preference but I do find the use of 'IntermediateType` somewhat confusing, when I read the code as it just does not register as a type parameter but rather a type itself.

Reply:
Type name changed to T.